src/diffusers/schedulers/scheduling_dpm_encoder.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2022 Stanford University Team and The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# DISCLAIMER: This code is strongly influenced by https://github.com/pesser/pytorch_diffusion
+# and https://github.com/hojonathanho/diffusion
+
+import math
+from dataclasses import dataclass
+from typing import Optional, Tuple, Union
+
+import numpy as np
+import torch
+
+from ..configuration_utils import ConfigMixin, register_to_config
+from ..utils import BaseOutput, deprecate
+from .scheduling_utils import SchedulerMixin
+from .scheduling_ddim import DDIMScheduler
+
+
+@dataclass
+class ReverseDDIMSchedulerOutput(BaseOutput):
+    """
+    Output class for the reverse scheduler's step function output.
+
+    Args:
+        next_sample (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)` for images):
+            Computed sample (x_{t+1}) of previous timestep. `next_sample` should be used as next model input in the
+            reverse denoising loop.
+        pred_original_sample (`torch.FloatTensor` of shape `(batch_size, num_channels, height, width)` for images):
+            The predicted denoised sample (x_{0}) based on the model output from the current timestep.
+            `pred_original_sample` can be used to preview progress or for guidance.
+    """
+
+    next_sample: Optional[torch.FloatTensor] = None
+    pred_original_sample: Optional[torch.FloatTensor] = None
+
+
+class DPMEncoderScheduler(DDIMScheduler):
```

Comment:
We don't abstract between schedulers - could you please copy-paste the functionality of `DDIMScheduler` here?